pyccel/ast/bitwise_operators.py

```diff
@@ -29,80 +29,137 @@
+    arg : TypedAstNode
+        The argument passed to the operator.
     """
     __slots__ = ()
     _precedence = 14
 
-    def _calculate_dtype(self, *args):
+    def _calculate_dtype(self, arg):
+        """
+        Get the dtype and precision.
+
+        Arguments must be integers or booleans. Any booleans are cast
+        to integers.
+
+        Parameters
+        ----------
+        arg : tuple
+            The argument passed to the operator.
+
+        Returns
+        -------
+        dtype : DataType
+            The  datatype of the result of the operation.
+        precision : integer
+            The precision of the result of the operation.
+        """
         dtype = NativeInteger()
-        a = args[0]
-        if a.dtype not in (NativeInteger(), NativeBool()):
-            raise TypeError('unsupported operand type(s): {}'.format(args))
+        if arg.dtype not in (NativeInteger(), NativeBool()):
+            raise TypeError(f'unsupported operand type(s): {arg}')
 
-        self._args      = (PythonInt(a) if a.dtype is NativeBool() else a,)
-        precision = a.precision
+        self._args      = (PythonInt(arg) if arg.dtype is NativeBool() else arg,)
+        precision = arg.precision
         return dtype, precision
 
     def __repr__(self):
-        return '~{}'.format(repr(self.args[0]))
+        return f'~{repr(self.args[0])}'
 
 #==============================================================================
 
 class PyccelBitOperator(PyccelOperator):
-    """ Abstract superclass representing a python
-    bitwise operator with two arguments
+    """
+    Abstract superclass representing a Python bitwise operator with two arguments.
+
+    Abstract superclass representing a Python bitwise operator with two arguments.
 
     Parameters
     ----------
-    arg1: TypedAstNode
-        The first argument passed to the operator
-    arg2: TypedAstNode
-        The second argument passed to the operator
+    arg1 : TypedAstNode
+        The first argument passed to the operator.
+    arg2 : TypedAstNode
+        The second argument passed to the operator.
     """
     _shape = None
     _rank  = 0
     _order = None
     __slots__ = ('_dtype','_precision')
 
+    def __init__(self, arg1, arg2):
+        super().__init__(arg1, arg2)
+
     def _set_order(self):
         pass
 
     def _calculate_dtype(self, *args):
-        """ Sets the dtype and precision
+        """
+        Get the dtype and precision.
 
-        If one argument is a string then all arguments must be strings
+        If one argument is a string then all arguments must be strings.
 
         If the arguments are numeric then the dtype and precision
-        match the broadest type and the largest precision
+        match the broadest type and the largest precision.
         e.g.
             1 + 2j -> PyccelAdd(LiteralInteger, LiteralComplex) -> complex
+
+        Parameters
+        ----------
+        *args : tuple
```

Comment:
This type information does not look useful to me. What the user (or developer) needs to know is that one or more instances of `TypedAstNode` is expected.

Reply:
Huh, not sure how I ended up writing this. It should be:
```suggestion
        *args : tuple of TypedAstNode
```